module/zfs/dsl_crypt.c

```diff
@@ -1452,11 +1453,14 @@ spa_keystore_change_key_sync_impl(uint64_t rddobj, uint64_t ddobj,
 	 * Stop recursing if this dsl dir didn't inherit from the root
 	 * or if this dd is a clone.
 	 */
-	VERIFY0(dsl_dir_get_encryption_root_ddobj(dd, &curr_rddobj));
-	if (!skip && (curr_rddobj != rddobj || dsl_dir_is_clone(dd))) {
+
+	ret = dsl_dir_get_encryption_root_ddobj(dd, &curr_rddobj);
```

Comment:
I don't know if it's a _huge_ concern, but it looks like there's a chance a non-zero or non-ENOENT return could get past the VERIFY check by triggering the return path here (I realized that with my own attempt at this after my initial fix).